src/lwc/psElevateTokenHandler/psElevateTokenHandler.js

```diff
@@ -93,16 +111,17 @@ class psElevateTokenHandler {
         let self = this;
 
         window.onmessage = async function (event) {
-
             if (self._visualforceOriginUrls) {
                 self._visualforceOrigin = self._visualforceOriginUrls.find(
                     origin => event.origin === origin.value
-                ).value;
-            }
-
-            if (self._visualforceOrigin) {
-                const message = JSON.parse(event.data);
-                component.handleMessage(message);
+                );
+                self._visualforceOrigin =
```

Comment:
In this section I think we could use a `shouldHandleMessage` function that returns a boolean indicating if we should call `component.handleMessage`. I think we can get window.onmessage function down to about three lines total.